### PR TITLE
chore: improve ossf scorecard's score

### LINKED ISSUE #714: improve repo's ossf scorecard's score
> Scorecard is an automated tool that assesses a number of important heuristics [("checks")](https://github.com/ossf/scorecard#scorecard-checks) associated with software security and assigns each check a score of 0-10. You can use these scores to understand specific areas to improve in order to strengthen the security posture of your project. You can also assess the risks that dependencies introduce, and make informed decisions about accepting these risks, evaluating alternative solutions, or working with the maintainers to make improvements.

We are currently scoring `5.3/10`:

[OpenSSF Scorecard
](https://securityscorecards.dev/viewer/?uri=github.com/kelektiv/node-cron)

Find below the checks we need to improve on and the associated remediation steps.

## Poor scoring checks

image

## How to improve check scores

- [x] **Fuzzing**
    - add fuzzing test cases. I've setup this on my `import-from-esm` package, see commit [9abddb5](https://github.com/sheerlox/import-from-esm/commit/9abddb52bfa0562c14aa9e9c79ce749ef25dcc1a) (note: it uses `ava` but there's also a `jest` plugin).

- [x] **SAST**
    - run CodeQL checks in CI/CD by following the instructions [here](https://github.com/github/codeql-action#usage).

- [x] **Security-Policy**
    - add `SECURITY.md` ([example](https://github.com/sheerlox/import-from-esm/blob/main/.github/SECURITY.md)).

- [x] **Token Permissions**
    - run [StepSecurity](https://app.stepsecurity.io/secureworkflow/kelektiv/node-cron/release.yml/main?enable=permissions) and apply recommended policies.

- [x] **Pinned-Dependencies**
    - [x] setup already done in https://github.com/kelektiv/node-cron/pull/683.
    - [x] the Renovate app needs to be installed to this repository from https://github.com/apps/renovate.

- [x] **Code Review**
    - this is already enforced, so the score will improve over time (reviewed changesets pushed to `main`).

- [x] **Vulnerabilities**
    - [x] setup already done in https://github.com/kelektiv/node-cron/pull/683.
    - [x] the Renovate app needs to be installed to this repository from https://github.com/apps/renovate.

### PR DESCRIPTION
**Pull request by sheerlox**
## Description

Please take a look at the related issue for more information.

## Related Issue

Fixes #714.

### CODE REVIEW

**Comment by sheerlox** [comment]
this requires #743 to be merged

File: tests/helpers/is_cron_error.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import { ExclusiveParametersError } from '../../src/errors';
+
+export function isCronError(error: unknown): boolean {
+	return (
+		error instanceof Error &&
+		(error instanceof ExclusiveParametersError ||
+			error.message === 'time must be an instance of CronTime.' ||
```
Comment (intcreator) [inline tests/helpers/is_cron_error.ts:7]: is this a list of all errors across the codebase? is this something we'll have to maintain and update every time we change the text of one of the errors?

Comment (sheerlox) [reply tests/helpers/is_cron_error.ts:7]: yes, indeed we would need to update that list when adding/updating errors. a workaround could be to create a `CronError` class we would use everywhere and simply check if the error is an instance of it.

Comment (intcreator) [reply tests/helpers/is_cron_error.ts:7]: that would be preferred. that way people can make text changes to errors without breaking the build. also would be good for the error on this test to say "make sure the relevant code is using an instance of `CronError`" so new contributors know how to fix it

**Review by intcreator — APPROVED** [review]
`CronError` looks great!

**Pull request merged**